Refactor uninstall controller for structured log

c3y1huang opened the pull request:
Refactor setting controller to address structured logging implementation.

https://github.com/longhorn/longhorn/issues/1733
https://github.com/longhorn/longhorn-manager/pull/661

joshimoo commented on controller/uninstall_controller.go:
```diff
@@ -470,17 +485,19 @@ func (c *UninstallController) deleteInstanceManagers(instanceManagers map[string
-			"type": "instanceManager",
-			"name": im.Name,
-		}
+		log := c.logger.WithFields(
```
Add a getLoggerForInstanceManager function to the instance-manager-controller, same idea as above for the engine.
The idea is to have a getLoggerX for each crd type.

joshimoo commented on controller/uninstall_controller.go:
```diff
@@ -491,50 +508,62 @@ func (c *UninstallController) deleteInstanceManagers(instanceManagers map[string
 }
 
 func (c *UninstallController) deleteManager() (bool, error) {
+	log := c.logger.WithFields(
```
For the DaemonSet/ReplicaSet you can add the getLoggerForX functions to this controller for now.
the idea is that the fieldName is `daemonSet`, with the value of the daemonsSet name.

joshimoo commented on controller/uninstall_controller.go:
```diff
@@ -551,54 +580,72 @@ func (c *UninstallController) deleteDriver() (bool, error) {
 	}
 	wait := false
 	for _, name := range deploymentsToClean {
+		log := c.logger.WithFields(
```
See above for deamonSet

joshimoo reviewed (changes requested):
Thanks for the pr, I added some suggestions.

c3y1huang commented:
> Thanks for the pr, I added some suggestions.

Thanks for the review, I've addressed all in https://github.com/longhorn/longhorn-manager/pull/706/commits/0cf8db6d12d8453bc601cdd320a27f1f4db6abaf.

joshimoo reviewed (approved):
LGTM, for the daemonset/deployment case you could have used references to the actual deployment/daemonset but since we currently only use this in the uninstall controller we can refactor that part later :)

joshimoo commented:
@c3y1huang Thanks for the PR and the addition of the new logger functions :)
